lib/project/pro_motion/data_table.rb

```diff
@@ -0,0 +1,187 @@
+module ProMotion
+  module DataTable
+    include ProMotion::Styling
+    include ProMotion::Table::Utils
+
+    def table_view
+      self.view
+    end
+
+    def screen_setup
+      set_up_reload_notification
+      set_up_fetch_controller
+
+      # TODO - implement dynamic row height
+      # set_up_row_height
+    end
+
+    def set_up_reload_notification
```

Comment:
As in all of them from this point below. Excepting the delegate methods which will need to be public, of course.